File: sort/bubbleSort/bubbleSort.py
data = [2, 1, 7, 5, 6, 3, 2, 0, 9]


def bubbleSort(data):
    n = len(data)
    # The out layer loop control times from head to end
    for j in range(n-1):
        # use count to record the counts about all exchange
        count = 0
        # the inner layer loop control times
        for i in range(0, n-1-j):
            if data[i] > data[i+1]:
                data[i], data[i+1] = data[i+1], data[i]
                count += 1
        if 0 == count:
            break
    return data


print(bubbleSort(data))

File: sort/bubbleSort/bubbleSort.php
<?php
/**
 * Date: 2019/9/16
 * Time: 11:05
 * @param array $data
 * @return array
 */

function bubbleSort(array $data)
{
    $count = count($data);
    for ($i = 1; $i < $count; $i++) {
        for ($j = 0; $j < $count - $i; $j++) {
            if ($data[$j] > $data[$j + 1]) {
                list($data[$j], $data[$j + 1]) = [$data[$j + 1], $data[$j]];
            }
        }
    }
    return $data;
}

print_r(bubbleSort([0, 2, 9, 1, 5, 6, 0, 3, 4]));